src/balloonpanel/balloonpanelview.js

```diff
@@ -0,0 +1,371 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals window, document, Range, HTMLElement */
+
+import View from '../view.js';
+import Template from '../template.js';
+
+const arrowLeftOffset = 30;
+const arrowTopOffset = 15;
+
+/**
+ * The balloon panel view class.
+ *
+ * See {@link ui.balloonPanel.BalloonPanel}.
+ *
+ * @memberOf ui.balloonPanel
+ * @extends ui.View
+ */
+export default class BalloonPanelView extends View {
+	/**
+	 * @inheritDoc
+	 */
+	constructor( locale ) {
+		super( locale );
+
+		const bind = this.bind;
+
+		this.template = new Template( {
+			tag: 'div',
+			attributes: {
+				class: [
+					'ck-balloon-panel',
+					'ck-link-balloon-panel',
+					bind.to( 'arrow', ( value ) => `ck-balloon-panel_arrow_${ value }` ),
+					bind.if( 'isVisible', 'ck-balloon-panel_visible' )
+				],
+
+				style: {
+					top: bind.to( 'top', ( value ) => `${ value }px` ),
+					left: bind.to( 'left', ( value ) => `${ value }px` ),
+					maxWidth: bind.to( 'maxWidth', ( value ) => `${ value }px` )
+				}
+			}
+		} );
+
+		this.register( 'content', el => el );
+
+		/**
+		 * Model of this balloon panel view.
+		 *
+		 * @member {ui.balloonPanel.BalloonPanelView} ui.balloonPanel.BalloonPanelView#model
+		 */
+	}
+
+	/**
+	 * Check if balloon is currently visible.
+	 *
+	 * @returns {Boolean}
+	 */
+	get isVisible() {
+		return this.model.isVisible;
+	}
+
+	/**
+	 * Show balloon and start listen to `keydown` event for closing by `Esc` key.
+	 */
+	show() {
+		this.model.isVisible = true;
+
+		// Attach keydown listener for closing panel on Esc press.
+		this._keyCloseCallback = evt => {
```

Comment:
I'd rather synchronise listener attachment with `change:isVisible` than with `show()/hide()`. What if some piece of code changes the model attribute directly? It would break the system because it would not attach/detach the listener. `show()/hide()` are nothing but syntax sugar here.